dapp_runner/runner/payload.py

```diff
@@ -13,6 +17,46 @@
 
 from .error import RunnerError
 
+logger = logging.getLogger(__name__)
+
+
+async def resolve_manifest(desc: PayloadDescriptor):
+    """Resolve a dynamically-generated manifest payload.
+
+    When the payload definition contains the `manifest_generate` key, it
+    attempts to construct a correctly constructed manifest based on a simplified
+    set of parameters and replaces the `manifest_generate` with a new `manifest` key.
+
+    Example payload:
+
+    ```yaml
+    payloads:
+        backend:
+            runtime: "vm/manifest"
+            params:
+                manifest_generate:
+                    image_hash: "e3c964343169d0a08b66751bfba89a90ec97544d8752c9a3e4ae0901"
+                    outbound_urls:
+                        - "http://bor.golem.network"
+                        - "https://geth.golem.network:55555"
+                        - "https://bor.golem.network"
+                        - "http://geth.testnet.golem.network:55555"
+    ```
+
+    """
+
+    if "manifest" not in desc.params and "manifest_generate" in desc.params:
+        generated_manifest_params = desc.params.pop("manifest_generate")
```

Comment:
```suggestion
        manifest_generate_params = desc.params.pop("manifest_generate")
```
As past tense is used too soon.